app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx

```diff
@@ -45,28 +50,40 @@ export class ExecutionMarker extends React.Component<IExecutionMarkerProps, IExe
   private handleStageClick = () => {
     ReactGA.event({category: 'Pipeline', action: 'Stage clicked (bar)'});
     this.props.onClick(this.props.stage.index);
-  }
+  };
 
   public render() {
-    const stage = this.props.stage;
+    const {stage, application, execution} = this.props;
     const markerClassName = [
       'clickable',
       'stage',
       'execution-marker',
       `stage-type-${stage.type.toLowerCase()}`,
       `execution-marker-${stage.status.toLowerCase()}`,
       this.props.active ? 'active' : '',
+      this.props.previousStageActive ? 'after-active' : '',
```

Comment:
If you're going through the trouble of decomposing props, decompose `previousStageActive` and `active` too.